### DISCUSSION TITLE
Could you provide an example on payload of Inline Equation?

### DISCUSSION THREAD

**Discussion by RickVincent** [jheddings/notional#178] [Q&A]
I am trying to build inline equation in a database, by method notion.pages.update(), but blocking by the error of the payload.

The expression of the equation is something like:

```
'\color{#CB345A}{text~need-to-be-colored-here}', or
'\color{#B55233}\text{text~need-to-be-colored-here}'
```

My current properties built as below.

```python
COLOR_EQUATION_PREFIX = r'\color{'
COLOR_EQUATION_SUFFIX = r'}\text{'
color_code = '#CB345A'
colored_color_text_name_expression = str(COLOR_EQUATION_PREFIX + color_code + COLOR_EQUATION_SUFFIX + color_text_name + '}')
props = {
                'title': color_text_name,
                'ColorName': types.EquationObject[types.RichTextObject[colored_color_text_name_expression]],
            }
```

Understood the payload has issue and doesn't comply with notional's API, so got below error.

```
pydantic.error_wrappers.ValidationError: 1 validation error for EquationObject
equation
  field required (type=value_error.missing)
```

could you provide an example on payload of the types.EquationObject[]? 

(Can't find similar example in the code. The orm.py showcases the types.ExternalFile[] usage, but yet to understand why after checked with notion.so API description.)

**Comment by jheddings** [comment]
HI @RickVincent - thanks for using notional!

Just to confirm, are you trying to create the equation as a page property (column) in your database, a block on the new page or as part of the rich text within another block?

**Reply by RickVincent** [reply]
Yes, creating the equation as a page property is exactly I am trying to implement. Just as below screenshot.
Looking for your help...

![image](https://user-images.githubusercontent.com/77380202/228562497-d27e8d7f-bfa1-4059-a59a-6cfeeea13d73.png)
